guide/docs/interactions/select-menus.mdx

```diff
@@ -1,8 +1,291 @@
+        super().__init__(
+            placeholder="Choose an animal",
+            min_values=1,
+            max_values=1,
+            options=options,
+        )
+
+    # this callback is called each time a user has selected an option
+    async def callback(self, inter: disnake.MessageInteraction):
+        # Use the interaction object to respond to the interaction
+        # `Self` refers to this StringSelect object, and the `values`
+        # attribute contains a list of the user's selected options
+        # We only want the first one
+        await inter.response.send_message(f"Your favorite type of animal is: {self.values[0]}")
+
+
+# Now that we have created the Select object with it's options and callback, we need to attach it to a
+# view that will be displayed to the user in the command response.
+# Views have a default of 180s before timing out and the bot will stop listening for those events
+# you may pass any float here, or `None` if you wish to remove the timeout. Note: If None is passed
+# this view will persist only until the bot is restarted.  If you wish to have persistent views,
+# consider using low-level or check out the persistent view example:
+# https://github.com/DisnakeDev/disnake/blob/master/examples/views/persistent.py
+class DropDownView(disnake.ui.View):
+    def __init__(self):
+        # You would pass a new `timeout=` if you wish to alter it, but
+        # we will leave it empty for this example so that it uses the default
+        super().__init__()
+
+        # now let's add the `StringSelect` object we created above to this view
+        self.add_item(AnimalDropdown())
+
+
+# Finally, we create a basic bot instance with a command that will utilize the created view and dropdown
+bot = commands.InteractionBot()
+
+
+@bot.listen()
+async def on_ready():
+    print(f"Logged in as {bot.user} (ID: {bot.user.id})\n------")
+
+
+@bot.slash_command()
+async def animals(inter: disnake.ApplicationCommandInteraction):
+    """Sends a message with our dropdown containing the animals"""
+
+    # create the view with our dropdown object
+    view = DropDownView()
+
+    # respond to the interaction with a message and our view
+    await inter.response.send_message("What is your favorite type of animal?", view=view)
+
+
+if __name__ == "__main__":
+    bot.run(os.getenv("BOT_TOKEN"))
+```
+
+</TabItem>
+<TabItem value="decorator" label="decorator.py">
+
+```python
+# instead of subclassing `disnake.ui.StringSelect`, this example shows how to use the
+# `@disnake.ui.string_select` decorator directly inside the view to create the dropdown
+# component.
+class AnimalView(disnake.ui.View):
+    def __init__(self):
+        super().__init__()
+
+        # rather than creating a large decorator, we can create the options for this component here
+        # this also works if you wish to pass a list of options to this view to create options dynamically
+        self.animal_callback.options = [
```

Comment:
Right.  If the options are static, I normally wouldn't do it this way, but I also thought that I'd show that it does work and would allow the use of dynamic options, even though they weren't being used with this specific example.   I have no issues with committing an altered version with the options inside the decorator

Reply:
That's completely fair. I think I'd still prefer to highlight the decorator-only solution first and foremost, and instead mention that you can also assign the options separately in a comment or info box; it's certainly useful to know regardless, but makes things more complex for beginners.